lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -3,21 +3,11 @@
 const { getOptionValue } = require('internal/options');
 const experimentalImportMetaResolve =
   getOptionValue('--experimental-import-meta-resolve');
-const {
-  PromisePrototypeThen,
-  PromiseReject,
-} = primordials;
 const asyncESM = require('internal/process/esm_loader');
 
 function createImportMetaResolve(defaultParentUrl) {
   return async function resolve(specifier, parentUrl = defaultParentUrl) {
-    return PromisePrototypeThen(
-      asyncESM.esmLoader.resolve(specifier, parentUrl),
-      ({ url }) => url,
-      (error) => (
-        error.code === 'ERR_UNSUPPORTED_DIR_IMPORT' ?
-          error.url : PromiseReject(error))
```

Comment:
We shouldn't drop this error handler, which is catching unsupported dir import and returning `error.url` when that error throws. Certainly would be nice to refactor that handling, but we should retain the behaviour...

Reply:
gah! thank you.